src/shadowsocks_config.ts

```diff
@@ -338,3 +338,63 @@ export const SIP002_URI = {
     return `ss://${userInfo}@${uriHost}:${port.data}/${queryString}${hash}`;
   },
 };
+
+export interface OnlineConfig {
+  // URL endpoint to retrieve a Shadowsocks configuration.
+  readonly url: string;
+  // Any additional configuration (e.g. `certFp`, `httpMethod`, etc.) may be stored here.
+  readonly extra: {[key: string]: string};
+}
+
+// Ref: https://github.com/shadowsocks/shadowsocks-org/issues/89
+export const SIP008_URI = {
```

Comment:
Should all the fields here also be readonly?

Reply:
this library exports objects and tsc creates the ts declaration file, so we can't set readonly properties here. we should define and expose interfaces instead - maybe in a follow up PR?